include/arch/arm/arch/64/mode/kernel/vspace.h

```diff
@@ -142,7 +142,8 @@ static inline exception_t performASIDPoolInvocation(asid_t asid, asid_pool_t *po
     return EXCEPTION_NONE;
 }
 
+#endif
+
 void increaseASIDBindCB(asid_t asid);
 void decreaseASIDBindCB(asid_t asid);
 
-#endif
```

Comment:
This does not look correct. I can see that the previous `ifdef` is wrong (if I have traced it correctly as `AARCH64_VSPACE_S2_START_L1`), but these functions should be under an SMMU guard and not be declared when SMMU is off.

Reply:
Ah yes, I will wrap these in a SMMU ifdef